crossplane/README.md

```diff
@@ -0,0 +1,146 @@
+
+- Provider-specific Operator
+- Provider-specific Custom Resource Definitions, usually
+`*.<provider-name>.crossplane.io`
+- Other Kubernetes resources
+
+### Stage 3 - Provider Configuration
+
+Just deploying Providers is not enough for them to be functional. They still
+need to be configured with passwords, secrets, access keys, API keys, and the
+like. Each Provider comes with its configuration CRD, usually
+`providerconfigs.<provider-name>.crossplane.io`.
+
+For example, the AWS provider will require a `ProviderConfig` similar to the
+one below to be able to perform reconciliation. `crossplane-system/aws-creds`
+The Secret should contain AWS Access Key.
+
+```yaml
+apiVersion: aws.crossplane.io/v1beta1
+kind: ProviderConfig
+metadata:
+  name: default
+spec:
+  credentials:
+    source: Secret
+    secretRef:
+      namespace: crossplane-system
+      name: aws-creds
+      key: creds
+```
+
+## Design proposals
+
+### 1. Installation
+
+#### Install Crossplane as part of the App collection
+
+This is useful if we want to install Crossplane App on every Management Cluster
+by default. The configuration will live in `giantswarm/config`. This still
+allows us to change available settings and Providers per Cloud Provider and per
+Management Cluster.
+
+#### Install Crossplane as HelmRelease in management-clusters-fleet
+
+This is useful if we want to limit Crossplane to be deployed on select
+Management Clusters. Creating a single `HelmRelease` resource allows us to
+easily set chart values.
+
+### 2. Handling configuration and secrets
+
+#### Giant Swarm stores customer secrets
+
+In this case, Giant Swarm is responsible for all three stages of deployment,
+including Provider Configuration. This means we need to store sensitive data
+(AWS Access Key, GCP Account Keyfile, Azure Service Principal Keys, etc.),
+support their encryption and decryption, and deliver them to Management
+Clusters.
+
+#### Giant Swarm manages Crossplane, the customer provides the keys
+
+In this case, Giant Swarm is responsible for stages one and two. Our goal is to
+deliver Crossplane and a set of Providers. It is up to the customer to create a
+`Secret` and `ProviderConfig`. The structure of both resources is simple and
+well-documented. It also makes sense given how the `ProviderConfig` definition
+varies, depending on the Provider.
+
+### 3. Providers
+
+#### The list of Providers is pre-defined and immutable
+
+We deploy a provider that best matches the Cloud Provider's infrastructure:
+- AWS <https://github.com/crossplane-contrib/provider-aws>
+- Azure <https://github.com/crossplane-contrib/provider-azure>
+- CAPA <https://github.com/crossplane-contrib/provider-aws>?
+- GCP <https://github.com/crossplane/provider-gc>
+- KVM is not supported; obviously, it's the customers' infrastructure
+- OpenStack TBD (<https://github.com/crossplane-contrib/provider-terraform>???)
+
+No other providers are available, even upon request. The offering is "managed".
```

Comment:
I believe we have to start with that, otherwise we're not gonna make the security right in time (nor should we, assuming we're switching towards customer-with-MC-admin pattern). This is the simplest and safest solution we can come up with in a limited time we have.